### PR TITLE
feature/ID-107-Create-a-mock-for-yoti-API-for-local-dev-use

### PR DESCRIPTION
**Pull request by fazemiah-moj**
# Purpose

Endpoints specified and payload/response structures set out and ready to be tested.

Some open questions/clarifications pending:

* Also docs do not make it clear where the **application_profile** sits within the session creation payload structure, which is an open question 
* Post validation - currently applying a generic pattern for matching UK ones. Is there a more specific Post Office version? how about international ones?
* Currently unclear what /idverify/v1/sessions/{sessionId}/instructions/pdf should actually return. Docs indicate this would be used to get and create the PDF letter, but how is this possible with no data sending back on a 200? Question added on our Word document

Fixes ###-####

## Approach

_Explain how your code addresses the purpose of the change._

## Learning

_Any tips and tricks, blog posts or tools which helped you._

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation where relevant
* [ ] I have added tests to prove my work
* [ ] I have run an accessibility tool against the changes and applied fixes
* [ ] The team have tested these changes

### CODE REVIEW

File: docker/yoti/data/branches.json
Comment (gregtyler) [inline docker/yoti/data/branches.json:1]: I don't think this file is used?

File: docker-compose.yml [hunk 1]
```diff
@@ -14,6 +14,20 @@ services:
     volumes:
       - ./docker/sirius:/opt/imposter/config
 
+  yoti-mock:
+    image: outofcoffee/imposter
+    restart: unless-stopped
+    ports:
+      - 8002:80
```
Comment (gregtyler) [inline docker-compose.yml:21]: No need expose ports under normal operation
```suggestion
```

File: docker/yoti/data/tsconfig.json
Comment (gregtyler) [inline docker/yoti/data/tsconfig.json:1]: We don't use TypeScript so shouldn't need a tsconfig file

File: docker/yoti/openapi.yml [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+                    - $ref: "#/components/schemas/AddressFormat2"
+                    - $ref: "#/components/schemas/AddressFormat3"
+                    - $ref: "#/components/schemas/AddressFormat4"
+
+    ID_DOCUMENT:
+      type: object
+      properties:
+        type:
+          type: string
+          enum: [ ID_DOCUMENT, SUPPLEMENTARY_DOCUMENT ]
```
Comment (gregtyler) [inline docker/yoti/openapi.yml:338]: ID_DOCUMENTs only have one option for type
```suggestion
          enum: [ ID_DOCUMENT ]
```

File: docker/yoti/openapi.yml [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+                      type: string
+                  document_types:
+                    type: array
+                    items:
+                      type: string
+                      enum: [PASSPORT, DRIVING_LICENCE, NATIONAL_ID, RESIDENCE_PERMIT]
+
+    SUPPLEMENTARY_DOCUMENT:
+      type: object
+      properties:
```
Comment (gregtyler) [inline docker/yoti/openapi.yml:363]: Should add `type` here too:
```suggestion
      properties:
        type:
          type: string
          enum: [ SUPPLEMENTARY_DOCUMENT ]
```

**Review by gregtyler — COMMENTED** [review]
A couple of minor comments but otherwise this looks really good based on the information we have

File: docker/yoti/openapi.yml (hunk 3 above)
Comment (fazemiah-moj) [reply docker/yoti/openapi.yml:363]: Ah missed that one, good spot :)

**Review by gregtyler — APPROVED** [review]

**Pull request merged**